### ISSUE TITLE
Config wizard adds extraneous required field for plateau LR Scheduler

### ISSUE THREAD

**Issue by nelson-liu** [allenai/allennlp#2420]
the `ReduceLROnPlateau` scheduler shouldn't require an optimizer, but its entry in the config wizard suggests that it does:

<img width="947" alt="screen shot 2019-01-22 at 1 01 57 pm" src="https://user-images.githubusercontent.com/7272031/51565110-eb411b00-1e45-11e9-81c7-a3e48cf43994.png">

Seems like the same category as #2402?

**Comment by joelgrus** [comment]
it seems like it does?

https://pytorch.org/docs/stable/optim.html#torch.optim.lr_scheduler.ReduceLROnPlateau

the config wizard doesn't really know what to do with pytorch types, because they're not type annotated

**Comment by nelson-liu** [comment]
Ah yeah, i meant the allennlp-wrapped version. i.e., there's no need to specify a key for optimizer here:

https://github.com/allenai/allennlp/blob/8ff832471364b09278d5863f31c47fa32662dd08/training_config/quarel_parser_elmo.json#L83-L88

Probably hard to fix, but would be nice to have / consider?

**Comment by joelgrus** [comment]
I see, yeah, that would require one-off logic somewhere

**Comment by matt-gardner** [comment]
This is actually related to the sequential dependencies issue, for the same reason that you wouldn't want the `Model` to take a `Vocabulary` specification in the config wizard, even though the `Model` has a `Vocabulary` argument.  I'm not at all sure how to handle this in a general way in the config wizard, though.
